VSP efficiency + retrieving efficiency directly

AngelaMarchi opened the pull request:
The modifications to the code solve the issue of Energy calculations for variable speed pumps #71. 

Changes to the getenergy function (in hydraulic.c) allow to compute the
correct efficiency of variable speed pumps according to the affinity
laws.
Changes to epanet2.h and epanet.c are for retrieving the efficiency of
pumps through the ENgetlinkvalue function (code 16 for efficiency)

eladsal commented:
Thanks @AngelaMarchi. Looks fine to me. Will wait for another per of eyes before we merge.

AngelaMarchi commented:
Thank you @eladsal and sorry for the late in doing this. Also, please, let me know if there are problems (I am not sure what the "All checks have failed" means).
Thank you again!

samhatchett commented:
Thanks @AngelaMarchi - I'm looking at this with @jamesuber this morning. In the meantime, would you mind tweaking the code style to match existing code? particularly, 

```
if {
  ...
}
else {
  ...
}
```

... and using spaces around the assignment of variables, i.e. `v = a` rather than `v=a`

samhatchett commented:
also, the failed tests are a reminder for us to fix the tests. sorry for the confusion.

jamesuber commented:
Thanks @AngelaMarchi ! The section of your commit -
```
		  if (Pump[j].Upat>0) { 
			  p = (Htime+Pstart)/Pstep;
			  temp = p % (long) Pattern[Pump[j].Upat].Length;
			  patMult = Pattern[Pump[j].Upat].F[temp]; 
		  } 
```
is used in lieu of the linkSetting[] when there is a utilization pattern. This seems to duplicate the code in https://github.com/OpenWaterAnalytics/EPANET/blob/master/src/hydraul.c#L627 that actually modifies the linkSetting. So from the perspective of not wanting the same code in two different places, do you know if this code is actually needed - i.e. that your code in getEnergy() can be called prior to the above referenced existing code in demands()? If so, then probably this is another small example where the code base should be designed a bit differently... not sure that's a valid discussion here cause presumably those larger efforts will be focused at epanet-dev (a.k.a. Epanet3). So if it certainly must be in two separate places, then probably just leave it in two separate places.

AngelaMarchi commented:
Thank you, @samhatchett : I have updated  the style (and sorry, I didn't pay attention to it before!)
Thank you also to @jamesuber for letting me know about the LinkSetting: I have updated the code, so only LinkSetting is used now (LinkSetting has the value of the pump pattern at each time step).
Please, let me know if there are problems.

samhatchett commented:
Thanks @AngelaMarchi - this looks good. we'll give this PR about 24 hours more in case others would like to comment.

AngelaMarchi commented:
I think I have been able to change the previous version:
- Thank you @eladsal : vb and bas files are updated and the code to retrieve the efficiency is improved
- all the previous suggestions from @samhatchett and @jamesuber are incorporated. 
Please, let me know if there are other problems and thanks to all for your help!

samhatchett commented:
looks like no further comments. Thanks so much @AngelaMarchi
